### PR TITLE
fix cmake atomic detection with clang and freebsd

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -635,7 +635,7 @@ if (NOT Windows)
 	set(ATOMICS_TEST_SOURCE [=[
 		#include <atomic>
 		#include <cstdint>
-		std::atomic<int> x{0};
+		std::atomic<int> x(0);
```
Comment (aldenml) [inline CMakeLists.txt:638]: hi @FranciscoPombal, there is something slightly different with AppleClang 12.0.0.12000032, however, I can't find any reference to it.

**Comment by aldenml** [comment]
Interesting it fails with `Clang 8.0.1`

**Comment by aldenml** [comment]
looks like a case of a missing `-std=c++17`

**Comment by aldenml** [comment]
Something is wrong with FreeBSD, maybe these are relevant:
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=230888
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=243364

**Comment by arvidn** [comment]
The cirrus CI broke boost-build a few days ago too. Did they update their image, or was there something else that caused this to start failing?

**Comment by aldenml** [comment]
I detected the problem when this PR (https://github.com/arvidn/libtorrent/pull/5748), the cmake atomic detection change landed in `master`. The problem is only for FreeBSD, because macOS is fine. I tried to change the image and other things with no positive result.

**Comment by arvidn** [comment]
@FranciscoPombal any idea?It looks like the logic to find libatomtc fails on FreeBSD.

**Comment by aldenml** [comment]
@arvidn, I see that you fixed freebsd in `RC_2_0`, I will wait for the merge in `master` to rebase this PR

**Comment by arvidn** [comment]
@aldenml the cirrus CI build still fails in the cmake step failing to find libatomic though, with this patch

**Comment by aldenml** [comment]
oh, I see that with your change to pin to `boost-1.74` in `RC_2_0` cirrus CI is green, I assumed that was it.

**Comment by arvidn** [comment]
yeah, for some reason this `libatomic` issue doesn't manifest in `RC_2_0`. Not sure why.

**Pull request closed without merging**